package/libpiksi/libpiksi/src/settings.c

```diff
@@ -1228,33 +204,19 @@ settings_ctx_t *settings_create(void)
     return ctx;
   }
 
-  /* Register standard types */
-  settings_type_t type;
+  ctx->setreg = setreg_create();
 
-  if (type_register(ctx, int_to_string, int_from_string, NULL, NULL, &type) != 0) {
-    destroy(&ctx);
-    return ctx;
-  }
-  assert(type == SETTINGS_TYPE_INT);
-
-  if (type_register(ctx, float_to_string, float_from_string, NULL, NULL, &type) != 0) {
-    destroy(&ctx);
-    return ctx;
-  }
-  assert(type == SETTINGS_TYPE_FLOAT);
-
-  if (type_register(ctx, str_to_string, str_from_string, NULL, NULL, &type) != 0) {
-    destroy(&ctx);
-    return ctx;
-  }
-  assert(type == SETTINGS_TYPE_STRING);
+  setreg_api_t api = {0};
```

Comment:
this could probably use a single list initializer:
```
setreg_api_t api = {
 .ctx = (void *)ctx,
 .send = send_wrap,
 .send_from = send_from_wrap,
 .wait = wait_wrap,
 .signal = signal_wrap,
 .register_cb = reg_cb_wrap,
 .unregister_cb = unreg_cb_wrap,
 .log = piksi_log,
};
```